### ISSUE TITLE
There are inconsistent and confusing postcss/autoprefixer warnings from "styles" in angular.json

### ISSUE THREAD

**Issue by rsc092020** [angular/angular-cli#24783]
### Command

build, serve

### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### Description

When seeing postcss/autoprefixer warnings from `ng serve` or `ng build`, the output of the where the warning is happening is hard to parse through. Also, the warnings do not consistently show up in the logs.

Regression details:
I know the filenames for the postcss/autoprefixer warnings used to display differently, so I would say that's a regression. I'm not sure whether that's directly related to angular-cli or a package it happens to depend on. 

As far as the issue with the warnings not always getting logged, I'm not sure which version this behavior changed.

I know that the behavior of both of these issues was different in @angular/cli 11. 

### Minimal Reproduction

repo: https://github.com/rsc092020/autoprefixer-warnings-14.git

steps:

- `ng new autoprefixer-warnings-14`

- Add css file, and update angular.json to include css file in "styles" array.

- run `ng serve` or `ng build` 

- Observe autoprefixer warning:

```
./styles2.css.webpack[javascript/auto]!=!./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[5].rules[0].oneOf[0].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[5].rules[0].oneOf[0].use[2]!./styles2.css - Warning: Module Warning (from ./node_modules/postcss-loader/dist/cjs.js):
Warning

(5:3) autoprefixer: You should write display: flex by final spec instead of display: box
```

- run `ng serve` or `ng build` again
- Observe that the build completed and no autoprefixer warning

It appears that i can clean the build cache to get the warning to show up again, but there are other warnings, like for commonjs dependencies, that will always show up. I don't understand why these warnings are different.

### Your Environment

```text
Angular CLI: 14.2.10
Node: 16.19.0
Package Manager: npm 8.19.3
OS: win32 x64

Angular: 14.2.12
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1402.10
@angular-devkit/build-angular   14.2.10
@angular-devkit/core            14.2.10
@angular-devkit/schematics      14.2.10
@angular/cli                    14.2.10
@schematics/angular             14.2.10
rxjs                            7.5.7
typescript                      4.7.4
```

**Comment by JoostK** [comment]
This has been addressed in v15 in https://github.com/angular/angular-cli/pull/24308, so it should no longer be a problem with latest Angular.

**Comment by rsc092020** [comment]
What about the other problem I mentioned, which is the warnings getting hidden in some builds. I think that is a bigger deal

**Comment by alan-agius4** [comment]
This is somewhat expected behaviour of Webpack, the reason being is that when the output of loaders is cached. They are not called again and hence the warnings are not added to the compilation.

On the other hand the commonjs warnings are displayed by the means of a plugin which is not cached.
